### PR TITLE
Capturing CTRL+D (EOF) to exit the shell

### PR DESCRIPTION
**Pull request by eubnara**
- support capturing CTRL+D to exit the shell
- add some tests for CTRL+C, CTRL+D cases

### CODE REVIEW

**Comment by eubnara** [comment]
With spring shell v1.2.0, `ctrl+d` exits the shell. I think this is the expected behavior.

**Comment by jvalkeal** [comment]
It's strange that `InteractiveShellRunnerTests` hangs up with jdk 8/11 but not with 17.

File: spring-shell-core/src/test/java/org/springframework/shell/jline/InteractiveShellRunnerTests.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+import org.jline.reader.LineReader;
+import org.jline.reader.LineReaderBuilder;
+import org.jline.terminal.Terminal;
+import org.jline.terminal.TerminalBuilder;
+import org.jline.utils.AttributedString;
+import org.jline.utils.AttributedStyle;
+import org.junit.jupiter.api.Test;
+import org.springframework.shell.ExitRequest;
+import static org.junit.jupiter.api.Assertions.*;
```
[the remark this answers is not in the document]
Comment (eubnara) [reply spring-shell-core/src/test/java/org/springframework/shell/jline/InteractiveShellRunnerTests.java]: @jvalkeal 
Thanks for feedback!
I changed it to use `assertj`.

**Comment by eubnara** [comment]
I currently use java 11.0.14 eclipse temurin installed by sdkman.

**Comment by jvalkeal** [comment]
I think this would be a good change and it did work when I ran a shell. We just can't merge it in this state as it jams a build. I briefly tried to debug it and it hangs on a thread somewhere inside jline classes.

**Comment by jvalkeal** [comment]
Great tests passes. I'll mark this to get merged.

**Pull request merged**